.github/workflows/continuous-integration.yml

```diff
@@ -68,6 +68,13 @@ jobs:
           BUILDTIME=$(date +%s)
           EOF
 
+      - name: Download coverage report
+        uses: actions/download-artifact@v2
+        if: ${{ github.ref == 'refs/heads/master' }}
+        with:
+          name: test-coverage-report
+          path: build/${{ matrix.buildtype }}/test-coverage-report.json
+
```

Comment:
The unit tests start the same time as the builds, so the coverage report isn't guaranteed to be available at this point. Why did we not end up pushing to S3 directly?

Reply:
Yes, I'm concerned about that, too. When we tried pushing to S3 directly, we tried creating our own directory so that we would not interfere with the build. But that directory did not have any CORS policy and so when my browser tried to access the file, I was getting a CORS error instead. I talked with @omgitsbillryan about this ([thread](https://dsva.slack.com/archives/CBU0KDSB1/p1626706583219900)) and he was thinking about what we could do to get a CORS policy on this directory, but we decided to go this route in order to get the work on this card done.

For right now, we are getting this file from the Jenkins deploy anyway, what would you recommend? I did some checking and the unit tests have been finishing in less time than the build, but you're right, it's not a guarantee. Should we comment this out for now? Add a second conditional to check for the file? Work with OPS more to get a CORS policy for a different directory?

Reply:
Reading through that thread, it makes that `vetsgov-website-builds-s3-upload` wasn't intended for public use. A couple of options:
1. Download the coverage artifact into the build path in the archive job. Then deploy would then push that up along with the build.
2. Direclty upload to the `apps.dev.va.gov` bucket from the unit test job. If you were to create a `testing-coverage` directory at the root, I don't think it would get overwritten (not 100% sure), since the deploy only syncs.

The first option is probably cleaner, but given the current build instabilities in GHA, you probably won't be able to validate it quickly. You'd also need to uncomment the archive job and exclude the actual deploy.

The second option can be validated earlier since it doesn't have a dependency on other jobs.

Reply:
I like option 1 in theory. It does seem cleaner, however by the time we hit the archive step the build is already in a tar file, which then gets "un-tarred" and pushed up through the `deploy.sh` script. Would we need to push this up to the deploy directory separately? That makes me a little nervous (I don't want things being inadvertently overwritten in deploy directories).

Option 2 seems a little safer, though there is the overwriting concern. We could try it and see what happens.

Any additional thoughts?

Reply:
For option 1, how much of the archive job can we uncomment? Are you saying we could try enabling the whole thing and then ditching the archive function in Jenkins? Or would both need to say?

Reply:
Option 2 does sound interesting too though. I'm not sure if I think it's more or less clean. I think I like the flow of option 2 better just because we'd be generating the coverage report and then immediately uploading it to a bucket.

Reply:
I forgot with option 1 that the build is already tarred before archiving, and it's just uploading the artifact directly to S3. I suppose it wouldn't be as good of an option now that I realize it.

Reply:
Then maybe we should try option 2 and see if we can create that directory and use the file from there.